examples/google-two-tier/main.tf

```diff
@@ -0,0 +1,83 @@
+# See https://cloud.google.com/compute/docs/load-balancing/network/example
+
+provider "google" {
+    region = "${var.region}"
+    project = "${var.project_name}"
+    account_file = "${file(var.account_file_path)}"
+}
+
+resource "google_compute_http_health_check" "default" {
+    name = "basic-check"
+    request_path = "/"
+    check_interval_sec = 1
+    healthy_threshold = 1
+    unhealthy_threshold = 10
+    timeout_sec = 1
+}
+
+resource "google_compute_target_pool" "default" {
+    name = "www-pool"
+    instances = ["${google_compute_instance.default.*.self_link}"]
+    health_checks = ["${google_compute_http_health_check.default.name}"]
+}
+
+resource "google_compute_forwarding_rule" "default" {
+    name = "www-rule"
+    target = "${google_compute_target_pool.default.self_link}"
+    port_range = "80"
+}
+
+resource "google_compute_instance" "default" {
+    count = 1
+
+    name = "tf-two-tier-example-${count.index}"
+    machine_type = "n1-standard-1"
+    zone = "${var.region_zone}"
+    tags = ["network-lb"]
+
+    disk {
+        image = "ubuntu-os-cloud/ubuntu-1204-precise-v20150625"
+    }
+
+    disk {
+        type = "local-ssd"
+        scratch = true
+    }
+
+    network_interface {
+        network = "default"
+        access_config {
+            # Ephemeral IP
+        }
+    }
+
+    metadata {
+        sshKeys = "ubuntu:${file(var.public_key_path)}"
```

Comment:
If you use startup-script you won't need this, but generally it's better to put keys in the project metadata so you can maintain them centrally.  

That would need to be part of the https://www.terraform.io/docs/providers/google/index.html docs

and users have hit it before: https://groups.google.com/forum/#!searchin/terraform-tool/ssh/terraform-tool/v4OdyUu0m4M/9oDdrxLAOpIJ